### ISSUE TITLE
Output files are getting overwritten when sample ids are not unique

### ISSUE THREAD

**Issue by famosab** [nf-core/sarek#1393]
### Description of the bug

When the "sampleID" in the samplesheet is not unique, only the output files for the final patient are kept. For example the following samplesheet would only keep the outout of the patient called 10perc.
```
patient,status,sample,lane,bam
5perc,0,normal,lane_1,imgag-benchmark/NA12877_21.bam
5perc,1,tumor,lane_1,imgag-benchmark/NA12878x3_23_NA12877_21_5.bam
10perc,0,normal,lane_1,imgag-benchmark/NA12877_21.bam
10perc,1,tumor,lane_1,imgag-benchmark/NA12878x3_23_NA12877_21_10.bam
```
When the samplesheet is altered, all output is kept:
```
patient,status,sample,lane,bam
5perc,0,normal_5perc,lane_1,imgag-benchmark/NA12877_21.bam
5perc,1,tumor_5perc,lane_1,imgag-benchmark/NA12878x3_23_NA12877_21_5.bam
10perc,0,normal_10perc,lane_1,imgag-benchmark/NA12877_21.bam
10perc,1,tumor_10perc,lane_1,imgag-benchmark/NA12878x3_23_NA12877_21_10.bam
```
The sarek runs throws no errors just the files are missing in the end.

### Command used and terminal output

```console
nextflow run nf-core/sarek -r 3.4.0 -profile cfc -work-dir work-bm -params-file params.json
```

### Relevant files

params.json:
```json
{
    "input": "/sfs/9/ws/paifb01-pm4onco/imgag-benchmark/benchmark_samplesheet.csv",
    "outdir": "/sfs/9/ws/paifb01-pm4onco/imgag-benchmark/results",
    "tools": "mutect2,strelka,manta,vep",
    "save_output_as_bam": true,
    "only_paired_variant_calling": true,
    "email": "famke.baeuerle@qbic.uni-tuebingen.de"
}
```

### System information

- Nextflow version: 23.10.1
- Hardware: HPC
- Executor: slurm
- Container engine: Singularity
- OS: Red Hat Enterprise Linux 8.8 (Ootpa)
- Version: 3.4.0

**Comment by asp8200** [comment]
Thanks for reporting this, @famosab. I feel like this situation should have been covered by plugin nf-validation. The version of nf-validation is not fixed in Sarek v3.4 (it will be in future versions). Which version of nf-validation was used in your run of Sarek?

**Comment by FriederikeHanssen** [comment]
I think it is the fact we only check for uniqueness with lane https://github.com/nf-core/sarek/blob/6aeac929c924ba382baa42a0fe969b4e0e753ca9/assets/schema_input.json#L48 but not individually.

**Comment by asp8200** [comment]
> I think it is the fact we only check for uniqueness with lane
> 
> https://github.com/nf-core/sarek/blob/6aeac929c924ba382baa42a0fe969b4e0e753ca9/assets/schema_input.json#L48
> 
> but not individually.

You reckon that we should also require that the `sample` is unique?

**Comment by FriederikeHanssen** [comment]
no, because you can have the same sample split over multiple lanes. I think we need to prevent overwriting. I suspect it is overwritten on the mapping level @famosab? Can you send `tree`of the offending output?

**Comment by famosab** [comment]
Here is the complete `tree` with all folders: [tree.txt](https://github.com/nf-core/sarek/files/14195784/tree.txt)
